AIO/src/org/aio/gui/Gui.java

```diff
@@ -223,14 +223,22 @@ public void close() {
     private TaskPanel addTask(final TaskType taskType) {
         TaskPanel taskPanel = TaskPanelFactory.createTaskPanel(taskType);
 
+        if (taskPanel == null){
+            throw new IllegalArgumentException(String.format("Task type %s not supported.", taskType.toString()));
+        }
+
+        String taskPanelKey = String.valueOf(taskPanel.hashCode());
```

Comment:
Any reason why you are converting this to a `String` rather than just storing as an `Integer` in the `Map`?

Reply:
Actually no, I started with a string but landed on the hashcode. I'll clean that up.